marimo/_plugins/ui/_impl/dataframes/dataframe.py

```diff
@@ -119,6 +121,10 @@ def __init__(
         self._transform_container = TransformsContainer(df)
         self._error: Optional[str] = None
 
+        if page_size not in range(
+            1, len(df) + 1
+        ):  # check if page_size is valid
+            page_size = min(len(df) + 1, 5)
```

Comment:
it might be ok to pass the page size for correctness and no need to clamp it. It's really a max size and the frontend should handle when it's above the df

Reply:
yeah, this is unnecessary